sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+        }
+
+        policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
+            buildConfiguration));
+        policies.add(new RequestIdPolicy());
+        policies.add(new AddHeadersPolicy(headers));
+        policies.add(new AddDatePolicy());
+
+        HttpPolicyProviders.addBeforeRetryPolicies(policies);
+
+        policies.add(retryPolicy == null ? DEFAULT_RETRY_POLICY : retryPolicy);
+
+        policies.addAll(this.policies);
+        HttpPolicyProviders.addAfterRetryPolicies(policies);
+        policies.add(new HttpLoggingPolicy(httpLogOptions));
+
+        // customized pipeline
+        HttpPipeline pipeline = new HttpPipelineBuilder()
+            .policies(policies.toArray(new HttpPipelinePolicy[0]))
+            .httpClient(httpClient)
+            .build();
+
+        TextAnalyticsClientImpl textAnalyticsAPI = new TextAnalyticsClientImplBuilder()
+            .endpoint(endpoint)
+            .pipeline(pipeline)
+            .build();
+
+        return new TextAnalyticsAsyncClient(textAnalyticsAPI, serviceVersion);
+    }
+
+    /**
+     * Set the default client option for one client.
+     *
+     * @param clientOptions TextAnalyticsClientOptions model that includes
+     * {@link TextAnalyticsClientOptions#getDefaultLanguage() default language} and
+     * {@link TextAnalyticsClientOptions#getDefaultCountryHint() default country hint}
+     * @return The updated TextAnalyticsClientBuilder object.
+     */
+    public TextAnalyticsClientBuilder clientOptions(TextAnalyticsClientOptions clientOptions) {
+        this.clientOptions = clientOptions;
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint for the Azure Text Analytics instance.
+     *
+     * @param endpoint The URL of the Azure Text Analytics instance service requests to and receive responses from.
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public TextAnalyticsClientBuilder endpoint(String endpoint) {
+        try {
+            new URL(endpoint);
+        } catch (MalformedURLException ex) {
+            throw logger.logExceptionAsWarning(new IllegalArgumentException("'endpoint' must be a valid URL", ex));
+        }
+        this.endpoint = endpoint;
+        return this;
+    }
+
+    /**
+     * Sets the credential to use when authenticating HTTP requests for this TextAnalyticsClientBuilder.
+     *
+     * @param subscriptionKey subscription key
+     * @return The updated TextAnalyticsClientBuilder object.
+     * @throws NullPointerException If {@code subscriptionKey} is {@code null}.
+     */
+    public TextAnalyticsClientBuilder subscriptionKey(String subscriptionKey) {
+        Objects.requireNonNull(subscriptionKey);
+
+        this.subscriptionKey = subscriptionKey;
+
+        // Clear TokenCredential in favor of subscription key credential
```

Comment:
I don't think we should have this side effect. Depending on the order that you invoke these methods tokenCredential or subscriptionKey, it'll change the behaviour of your authorization. If someone sets TokenCredential, you use it first to create the pipeline.